### PR TITLE
feat(ids-api): Enable getting fake users in v3 client. Configure for ids-api envs.

### PR DESCRIPTION
**Pull request by valurefugl**
## What

Add a parameter to the national registry v3 client that indicates that we want to try the gervimenn api if no user found in the main api.

Provide this parameter when client is used in ids-api, based on environment setting (only set in DEV).

## Why

The main api on DEV only provides a subset of the Gervimenn we are using.

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Introduced a new environment variable `ALSO_USE_FAKE_USER_API` across various configurations (dev, staging, prod).
  - Enhanced API behavior by allowing conditional use of a fake API based on configuration settings.

- **Bug Fixes**
  - Improved handling of API calls in the `getAllDataIndividual` method to ensure proper fallback to the fake API when needed.

- **Chores**
  - Adjusted resource limits and scaling configurations for improved performance and reliability across services.

### CODE REVIEW

**Review by magnearun — APPROVED** [review]

File: apps/services/auth/ids-api/src/app/user-profile/user-profile.service.ts [hunk 1]
```diff
@@ -45,7 +47,14 @@ export class UserProfileService {
...
     @Inject(LOGGER_PROVIDER)
     private logger: Logger,
-  ) {}
+    @Optional()
+    readonly configService?: ConfigService,
+  ) {
+    if (configService) {
+      this.alsoUseFakeApi =
+        configService.get<boolean>('ALSO_USE_FAKE_USER_API') ?? false
+    }
```
Comment (saevarma) [inline apps/services/auth/ids-api/src/app/user-profile/user-profile.service.ts:56]: You could do and remove L#34
```suggestion
      this.alsoUseFakeApi =
        configService?.get<boolean>('ALSO_USE_FAKE_USER_API') ?? false
```

File: apps/services/auth/ids-api/infra/ids-api.ts [hunk 2]
```diff
@@ -100,6 +100,11 @@ export const serviceSetup = (): ServiceBuilder<'services-auth-ids-api'> => {
         staging: 'digitaliceland',
         dev: 'digitaliceland',
       },
+      ALSO_USE_FAKE_USER_API: {
+        dev: 'true',
+        staging: 'false',
+        prod: 'false',
```
Comment (saevarma) [inline apps/services/auth/ids-api/infra/ids-api.ts:106]: How will this affect the fake user login in production used for the app review, where we enable a single fake user with random phone number using the env variable?

File: apps/services/auth/ids-api/src/app/app.module.ts [hunk 3]
```diff
@@ -56,6 +57,9 @@ import { ZendeskServiceConfig } from '@island.is/clients/zendesk'
     NotificationsModule,
     LoginRestrictionsModule,
     PasskeysModule,
+    NestConfigModule.forRoot({
```
Comment (saevarma) [inline apps/services/auth/ids-api/src/app/app.module.ts:60]: Why is this necessary?

File: libs/auth-api-lib/src/lib/delegations/alive-status.service.ts [hunk 4]
```diff
@@ -20,13 +21,22 @@ const decesead = 'LÉST'
...
     @Inject(LOGGER_PROVIDER)
     private logger: Logger,
-  ) {}
+    @Optional()
+    readonly configService?: ConfigService,
+  ) {
+    if (configService) {
+      this.alsoUseFakeApi =
+        configService.get<boolean>('ALSO_USE_FAKE_USER_API') ?? false
+    }
```
Comment (saevarma) [inline libs/auth-api-lib/src/lib/delegations/alive-status.service.ts:38]: Same like in the other service and remove L#24

**Review by thorkellmani — APPROVED** [review]
LGTM

File: apps/services/auth/ids-api/infra/ids-api.ts (hunk 2 above)
Comment (valurefugl) [reply apps/services/auth/ids-api/infra/ids-api.ts:106]: The idea is that these changes should not affect production at all.

In the env var we can also provide the name of the fake user and it is used as fallback.

File: apps/services/auth/ids-api/src/app/app.module.ts (hunk 3 above)
Comment (valurefugl) [reply apps/services/auth/ids-api/src/app/app.module.ts]: It is not. Removed.

File: apps/services/auth/ids-api/src/app/user-profile/user-profile.service.ts (hunk 1 above)
Comment (valurefugl) [reply apps/services/auth/ids-api/src/app/user-profile/user-profile.service.ts]: Fixed.

File: libs/auth-api-lib/src/lib/delegations/alive-status.service.ts (hunk 4 above)
Comment (valurefugl) [reply libs/auth-api-lib/src/lib/delegations/alive-status.service.ts]: Fixed.

File: apps/services/auth/ids-api/src/app/user-profile/user-profile.service.ts [hunk 5]
```diff
@@ -45,7 +46,13 @@ export class UserProfileService {
     private companyRegistryApi: CompanyRegistryClientService,
     @Inject(LOGGER_PROVIDER)
     private logger: Logger,
-  ) {}
+    @Optional()
+    private alsoUseFakeApi: boolean,
```
Comment (saevarma) [inline apps/services/auth/ids-api/src/app/user-profile/user-profile.service.ts:50]: Ah sorry we need the `alsoUseFakeApi` member definition on the class, but we can skip assigning it a default value as the optional chaining in the constructor will handle that.

So add L#34 again as `private alsoUseFakeApi: boolean` and remove L#50

Also `@Optional()` decorator was probably meant for the `configService`?

File: libs/auth-api-lib/src/lib/delegations/alive-status.service.ts [hunk 6]
```diff
@@ -26,7 +27,13 @@ export class AliveStatusService {
     private readonly companyRegistryClient: CompanyRegistryClientService,
     @Inject(LOGGER_PROVIDER)
     private logger: Logger,
-  ) {}
+    private alsoUseFakeApi: boolean,
```
Comment (saevarma) [inline libs/auth-api-lib/src/lib/delegations/alive-status.service.ts:30]: Same here

**Review by saevarma — APPROVED** [review]

File: apps/services/auth/ids-api/src/app/user-profile/user-profile.service.ts (hunk 5 above)
Comment (valurefugl) [reply apps/services/auth/ids-api/src/app/user-profile/user-profile.service.ts]: Fixed.

**Review by robertaandersen — APPROVED** [review]
LGTM

**Pull request merged**